### DISCUSSION TITLE
New Feature and 12v settings

### DISCUSSION THREAD

**Discussion by tobes49** [DavidBertet/PowerJeep#15] [Ideas]
Hello,  

Just found you project and looks exactly what I need for my kids cars. 

My car runs 12v 30000rpm motors and uses 3s Lipo for power, this has burnt out the standard 12v electonics twice now,   I have changed to direct relays but this is too powerful (the car pulls wheelies and causes the front to become so light when pulling away there no steering!)

I have a couple of questions?
Will changing the 60% in the motors file to 100% allow me to provide the full power to the motors?

Also as my 6 year old and 3 year old both use the car I would like to keep the high low power switch and have the output speed switchable?  eg full =100% and low speed is 50% etc
 
The other thing (but not that bothered) could we have a drag mode where the speed ramp up is much more aggressive?

Thank you again for the great work and my ESP and step down regulator are on order, cant wait to try it.

**Comment by tobes49** [comment]
After getting my ESP32 uploading the sketch, I see you can change the power by the webpage.  Would be nice to have the option on a switch.  Looking forward to getting it all fitted and working can add the bits later.

**Comment by tobes49** [comment]
After hitting some issues.  Which was narrowed down to the forward/reverse switch gave too much interference on the 3.3volt lines.
This was solved by running 12v  to the forward and reverse switch and using a change over relay to handle the forward/backwards switching this also simplified my wiring (will post a diagram later)

It now working correctly. I've still got make a mount for the throttle pedal and do some testing (worried about over heating but heating tape on to a slab of aluminium should help dissipate the heat)

![20250220_213210](https://github.com/user-attachments/assets/c8647156-35db-4435-a226-4c9f223a6d5b)

**Reply by tobes49** [reply]
Upgrades for future that will probably never happen!
The power wire is an old APM power module (Drone) that I had laying around, this can detect current and voltage which could be fed the ESP to give extra info

Wire throttle pedal direct to esp and have switching done using another pin on the ESP

Have a drag mode that ramps the speed up quicker

Control lights and indicators and voltage meter on dashboard.

But thank you for all the work you have put into this project 🙏

**Reply by DavidBertet** [reply]
Hey, thanks for the updates!

Some of it actually happened! I just pushed a new BETA version with the switch, the drag mode and separated throttle option.
https://github.com/DavidBertet/PowerJeep/tree/v2-beta

This is still an early version, but it's been working great so far. There is a demo of what the interface [looks like here](https://davidbertet.github.io/PowerJeep)

**Reply by tobes49** [reply]
Thank you for the update.  I'm currently on it and will try it soon.
![20250806_211109](https://github.com/user-attachments/assets/7cad92fa-140e-45eb-9400-264ddba01243)

**Reply by DavidBertet** [reply]
Awesome! Let me know how the board update worked, I admit having only the same dev boards and couldn’t realy get it a try!

**Reply by tobes49** [reply]
Ive hit some problems (some might be me as not the most advanced linux user)

git clone appears not to download all the files needed - (just downloaded as zip and got them this way)

when building formware it comes up with error:
```
✗ Firmware build failed                                                    	 
✗ Build error:
src/main.c: In function 'app_main':
src/main.c:51:33: error: 'panic_emergency_shutdown_hook' undeclared (first use in this function)
   51 |   esp_register_shutdown_handler(panic_emergency_shutdown_hook);
  	|                             	^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/main.c:51:33: note: each undeclared identifier is reported only once for each function it appears in
*** [.pio/build/esp32doit-devkit-v1/src/main.c.o] Error 1
========================= [FAILED] Took 216.36 seconds =========================
Environment      	Status	Duration
-------------------  --------  ------------
esp32doit-devkit-v1  FAILED	00:03:36.361
==================== 1 failed, 0 succeeded in 00:03:36.361 ====================
```

**Reply by tobes49** [reply]
Ive tried in Windows with GitHub addon and this seems to have uploaded it all to the ESP32.

Will try it in the car when I get home tonight,
